projects/map/src/lib/openlayers-map/tools/open-layers-map-click-tool.ts

```diff
@@ -28,17 +30,26 @@ export class OpenLayersMapClickTool implements MapClickToolModel {
 
   public enable(): void {
     this.enabled = new Subject();
-    OpenLayersEventManager.onMapClick$()
-      .pipe(takeUntil(this.enabled))
-      .subscribe(evt => {
-        const { scale, resolution } = OpenLayersHelper.getResolutionAndScale(evt.map.getView());
+    this.in3D$.pipe(
```

Comment:
Would rather move this to the `CesiumEventManager` code. Check there if you are in 3d or not, return `undefined` or `null` when not in 3d and the actual event if you are. Same for `OpenLayersEventManager`, there here it can be a simple check and then there is also no need to pass the 3d$ observable as constructor param